form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -36,6 +42,36 @@ class FormFlowService(
         return formFlowDefinitionRepository.getById(formFlowDefinitionId)
     }
 
+    fun open(formFlowInstance: FormFlowInstance) {
```

Comment:
This should probably be a either a helper class, or inside of the (definition) step itself, so the instance can ask the definition how onOpen should be resolved. Regardless, the open() should be performed in the instance, and not in the FormFlowService.

Reply:
How will implementation projects be able to extend or change the 'open' logic, if its in the definition

Reply:
They don't. They're not supposed to.

Reply:
I looked it up. There is an interesting talk about this exact topic: https://www.youtube.com/watch?v=US8QG9I1XW0&t=1994s

```
The more behavior you add to the class, the more behavior it also attracts. So you'd end up with a couple of very big classes ... The classes attract more and more behavior and become less and less cohesive. We end up calling those classes 'god classes'. That is a well-known anti pattern.
```